lib/rucio/web/rest/flaskapi/v1/common.py

```diff
@@ -192,7 +191,43 @@ def parse_scope_name(scope_name, vo):
     return scope_name.group(1, 2)
 
 
-def try_stream(generator, content_type=None) -> "flask.Response":
+def _cut_json_stream(generator: typing.Iterator[typing.AnyStr]) -> typing.Iterator[typing.AnyStr]:
+    """
+    Generator for cutting off JSON document endings until
+    the next element from the generator arrived.
+    """
+    last_document_end = None
+    for part in generator:
+        is_json_doc = False
+        document_end_index = -1
+        if isinstance(part, bytes):
+            json_document_end = b'}'
+            is_json_doc = part.rstrip().endswith(json_document_end)
+            if is_json_doc:
+                document_end_index = part.rfind(json_document_end)
+        elif isinstance(part, str):
+            json_document_end = '}'
+            is_json_doc = part.rstrip().endswith(json_document_end)
+            if is_json_doc:
+                document_end_index = part.rfind(json_document_end)
+
+        if is_json_doc:
+            if last_document_end:
+                yield last_document_end + part[:document_end_index]
+            else:
+                yield part[:document_end_index]
+            last_document_end = part[document_end_index:]
+        else:
+            if last_document_end:
+                yield last_document_end + part
+            else:
+                yield part
+
+    if last_document_end:
+        yield last_document_end
```

Comment:
Okay, I have a **hard** time understanding this. My first question is:
`is_json_doc` is set to a boolean value if the current string (without whitespaces) ends with a `}`. If it does (`is_json_doc` is `True`) the  `document_end_index` is set to the index of this position. Doesn't that mean that `last_document_end` is only whitespaces? Thus, why do we need it? The whitespaces after a json document should be fairly insignificant...